src/ray/core_worker/core_worker.cc

```diff
@@ -763,7 +791,11 @@ void CoreWorker::Shutdown() {
 
   // Now that gcs_client is not used within io service, we can reset the pointer and clean
   // it up.
-  gcs_client_.reset();
+  if (gcs_client_) {
```

Comment:
If I were you, I would place the disconnection logic into `gcs_client_`'s deleter.

Reply:
I'll leave it for a separate PR.

Reply:
If you plan to do that, maybe leave a TODO in case we forget?
Feel free to leave it under my name.